### PR TITLE
[TR] PIM-3979: Copy from product working copy

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-3979

| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | Y
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | -
| Scenarios pass?      | 
| Checkstyle issues?*  | N
| PMD issues?**        | -
| Changelog updated?   | -
| Fixed tickets        | -
| DB schema updated?   | N
| Migration script?    | -
| Doc PR               |

EE PR : https://github.com/akeneo/pim-enterprise-dev/pull/924

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js [hunk 1]
```diff
@@ -68,9 +66,6 @@ define([
             setField: function (field) {
                 this.field = field;
             },
-            selectionChanged: function (event) {
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js]: why did you remove that ?

Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy-field.js]: Because the clic event bubbles to the global div and matches the other listener, it's useless to have two listeners.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js [hunk 2]
```diff
@@ -80,106 +85,159 @@ define(
...
+            /**
+             * Render the copy element inside each field that can be copied
+             *
+             * @param {Object} event
+             */
+            addFieldExtension: function (event) {
+                var field = event.field;
+                var copyField = this.getCopyField(field);
+                if (copyField) {
+                    field.addElement('comparison', this.code, copyField);
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js]: Nice, you didn't forget the `this.code`

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js [hunk 3]
```diff
@@ -80,106 +85,159 @@ define(
...
-                this.render();
+                this.trigger('copy:context:change');
             },
+
+            /**
+             * Change the locale for copy context
+             *
+             * @param {String} locale
+             */
             setLocale: function (locale) {
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js]: We should do like attributes.js: use events for that. Dunno if we should do that now

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js [hunk 4]
```diff
@@ -80,106 +86,177 @@ define(
...
             },
+
+            /**
+             * Indicate if the specified field can be copied
+             *
+             * @param {Field} field
+             * @returns {Promise}
+             */
+            canBeCopied: function (field) {
+                return $.Deferred().resolve(
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js]: in one line ?

**Comment by sumbobyboys** [comment]
Awesome PR, good job !

File: src/Pim/Bundle/UIBundle/Resources/public/css/pim.less [hunk 5]
```diff
@@ -710,7 +710,7 @@ h3.tab-header {
 
 .comparison-mode {
   .comparison-elements-container {
-    .display-flex();
```
Comment (nidup) [inline src/Pim/Bundle/UIBundle/Resources/public/css/pim.less]: :flex:

**Comment by nidup** [comment]
GTM for me but i maybe miss/not understand all the consequences
